x-pack/plugin/security/qa/security-trial/src/javaRestTest/java/org/elasticsearch/xpack/security/apikey/ApiKeyRestIT.java

```diff
@@ -2112,6 +2115,270 @@ public void testWorkflowsRestrictionValidation() throws IOException {
+                ]
+              },
+              "certificate_identity": "%s"
+            }""", certificateIdentity));
+
+        setUserForRequest(createRequest, MANAGE_SECURITY_USER, END_USER_PASSWORD);
+        final ObjectPath createResponse = assertOKAndCreateObjectPath(client().performRequest(createRequest));
+        final String apiKeyId = createResponse.evaluate("id");
+
+        final ObjectPath fetchResponse = fetchCrossClusterApiKeyById(apiKeyId);
+
+        assertThat(fetchResponse.evaluate("api_keys.0.certificate_identity"), equalTo(certificateIdentity));
+    }
+
+    public void testUpdateCrossClusterApiKeyToAddCertificateIdentity() throws IOException {
+        final Request createRequest = new Request("POST", "/_security/cross_cluster/api_key");
+        createRequest.setJsonEntity("""
+            {
+              "name": "cross-cluster-key-update-test",
+              "access": {
+                "search": [
+                  {
+                    "names": [ "random" ]
+                  }
+                ]
+              }
+            }""");
+        setUserForRequest(createRequest, MANAGE_SECURITY_USER, END_USER_PASSWORD);
+        final String apiKeyId = assertOKAndCreateObjectPath(client().performRequest(createRequest)).evaluate("id");
+
+        final String certificateIdentity = "CN=updated-host,OU=engineering,DC=example,DC=com";
+        final Request updateRequest = new Request("PUT", "/_security/cross_cluster/api_key/" + apiKeyId);
+        updateRequest.setJsonEntity(Strings.format("""
+            {
+              "certificate_identity": "%s"
+            }""", certificateIdentity));
+        setUserForRequest(updateRequest, MANAGE_SECURITY_USER, END_USER_PASSWORD);
+        final ObjectPath updateResponse = assertOKAndCreateObjectPath(client().performRequest(updateRequest));
+        assertThat(updateResponse.evaluate("updated"), is(true));
+
+        final ObjectPath fetchResponse = fetchCrossClusterApiKeyById(apiKeyId);
+        assertThat(fetchResponse.evaluate("api_keys.0.certificate_identity"), equalTo(certificateIdentity));
+    }
+
+    public void testUpdateCrossClusterApiKeyToChangeCertificateIdentity() throws IOException {
+        final String originalCertIdentity = "CN=original-host,OU=engineering,DC=example,DC=com";
+        final Request createRequest = new Request("POST", "/_security/cross_cluster/api_key");
+        createRequest.setJsonEntity(Strings.format("""
+            {
+              "name": "cross-cluster-key-change-cert",
+              "access": {
+                "search": [
+                  {
+                    "names": [ "metrics" ]
+                  }
+                ]
+              },
+              "certificate_identity": "%s"
+            }""", originalCertIdentity));
+        setUserForRequest(createRequest, MANAGE_SECURITY_USER, END_USER_PASSWORD);
+        final String apiKeyId = assertOKAndCreateObjectPath(client().performRequest(createRequest)).evaluate("id");
+
```

Comment:
You could add an assert here to make sure it's the `originalCertIdentity` even though it's tested elsewhere.